lib/hotel.rb

```diff
@@ -0,0 +1,273 @@
+
+module My_Hotel
+  #to change room prices, change the values of ROOMS
+  ROOMS = {1 => 200,
+    2 => 200,
+    3 => 200,
+    4 => 200,
+    5 => 200,
+    6 => 200,
+    7 => 200,
+    8 => 200,
+    9 => 200,
+    10 => 200,
+    11 => 200,
+    12 => 200,
+    13 => 200,
+    14 => 200,
+    15 => 200,
+    16 => 200,
+    17 => 200,
+    18 => 200,
+    19 => 200,
+    20 => 200
+  }
+  class Hotel #keeps track of all the reservations and all the blocks.
+    attr_reader :rooms, :all_reservations, :all_blocks
+
+    def initialize
+      @rooms = ROOMS
+      @all_reservations = []
+      @all_blocks =[]
+    end
+
+    # can be used for "As an administrator, I can access the list of
+    #all of the rooms in the hotel"
+    def display_rooms
+      ROOMS.keys
+    end
+
+    #Can be used for "As an administrator, I can get the total cost
+    # for a given reservation" Or, you could just do reservation.cost
+    def get_cost(reservation)
+      reservation.cost
+    end
+
+    def date_range(first_night, last_night)
+      if first_night != last_night
+        @nights = (first_night..last_night)
+      else
```

Comment:
I'm not sure you want to use an instance variable here - that will set `@nights` for the entire `Hotel`. Instead, creating and returning a local variable is probably the right choice.

Reply:
This method would be a great place to do some error checking on the range of dates.